docs/section-3-creating-parsers.md

```diff
@@ -911,6 +911,38 @@ In case of some keywords defined in the `externals` array in a rule referencing
 
+## Releasing Grammars
+
+Once you are happy with the state of your grammar, you should consider releasing it for other people to use. As [mentioned above](#command-generate), the `tree-sitter generate` command creates _bindings_ for your grammar in a number of languages. This means that language support in Tree-sitter is a _matrix_, since you have to consider both the language that you want to parse and the language that you want to write your analysis tool in. As the author of a grammar, you should strive to release grammar packages for each of the languages that `tree-sitter generate` creates bindings for.
+
+To make this job easier, many grammar authors have set up CI jobs that will automatically publish new binding package releases whenever you push a new git tag to the grammar repository. If you are hosting your grammar repository on GitHub, you can use [this Actions workflow][python autorelease] from the Python grammar repository as a template. (Note that you will have to add secrets to your CI configuration so that it has permission to publish the new packages to the binding language's package registry.) With this kind of CI job in place, to release a new grammar version you would:
+
+1. Update any hard-coded grammar versions wherever they may appear in the grammar repository.
+2. Create and merge PR with those changes.
+3. Create and push a new git tag.
+4. Watch the ✨magic✨ happen.
+
+### A note on versions
+
+Implicit in the above is that you have to start applying version numbers to your grammar, which means you must decide on a strategy for assigning version numbers.
+
+In the early days of the Tree-sitter project, the only grammars that were available were written by the Tree-sitter maintainers themselves. New versions of the grammar packages were released in lockstep with new versions of Tree-sitter itself, and all of the packages were given the same version numbers. So in a Rust tool for analyzing Python code, you might have see a dependency on both `tree-sitter = 0.18.1` and `tree-sitter-python = 0.18.1`.
+
+Over time, we came to realize that this is not a helpful versioning scheme for the grammars. Tree-sitter provides an [ABI versioning mechanism][abi], so that the Tree-sitter version that you use at runtime does not have to be exactly the same as the Tree-sitter version used to generate the parser library. That means that grammar packages do not need to depend on specific versions of Tree-sitter.
+
+Moreover, by using the same version number for Tree-sitter and for the grammars, we were not communicating any useful information about the content of the grammar itself. Consumers of a grammar care most about which language constructs it is able to parse, and what shape of parse tree is produced as a result. When either of these things change, downstream consumers might have to update how they are using your grammar. _That_ is the information that we should communicate in a grammar version.
```

Comment:
The history lesson feels a bit excessive IMO. I think we should just focus on how things are now.